### DISCUSSION TITLE
Weekly Blog Post 5

### DISCUSSION THREAD

**Discussion by LPkkjHD** [SE-TINF22B6/Underwatch#55] [Announcements]
Hello and welcome to our weekly blog post in the fifth edition.
This week we are covering the topics of the EER, class diagrams as well as a small progress update on the project.

## EER Model 📀
We are using a relational database - as it is required by the lecture - to store the scores of the players. As we are planning to create an arcade game we don't need any fancy user accounting or session storage. The current flow for adding a new score is the following:
![](https://user-images.githubusercontent.com/87950162/282827200-a35b8dd1-c23d-42a5-bfef-610a70f91874.svg)
Following this we only have a singular table the EER is rather trivial and looks like this:
![JOqzheD038JxFOKly2v0nG5UH_h8sRM8XNzMkwP0KUwUW1H1Dnv_cc_iWIrMbHHU8HmU7k4AXHlbOBVK5xQA3HM4ZMAh1DxYwIbbmTvXoju8NWdW3xWCm1P7he8RDKEb0QPeRFDHBopIJwgWuao5KE8iWvLweDO1RhzuFFhkEhdiRDUjUZbcLBe7x_P__xhVQIGhgye7](https://github.com/SE-TINF22B6/Underwatch/assets/11832982/79154041-6d3c-4202-aa26-5807d027459d)

## Class Diagram 🕹️
For the game, which is a Java project creating an EER does not make any sense as the EER is for describing the structure. In order to create some structure for the further development as well as a proper documentation we created a class diagram to showcase how things work under the hood.
In the current form it looks like this:
![class_diagramm_game_short](https://github.com/SE-TINF22B6/Underwatch/assets/87950162/ceab9041-d373-422c-ad32-87a4a0e30853)

## Component Tree
In our Website the library we use is React. As React has components we are building a schema - the so called component tree.
It looks like this:
![SoWkIImgAStDuKhEIImkLd0iAF3CIqcjW7BzyalSyalpKg3SuEJyejIa_CGYc41uPdOcbEcSaPXRa9cVLym5CqD1DqNN3aaFgWHOBweGLZEAWE9mPpYavgK07Gi0](https://github.com/SE-TINF22B6/Underwatch/assets/11832982/7fab9880-2f4a-4e8e-b5c5-5993c196b296)

## Game Updates

Last but not least we also made some new progress in the game. We are now working with [Box2D](https://libgdx.com/wiki/extensions/physics/box2d) to have collisions. In addition the wrapper library [Box2Dlights](https://github.com/libgdx/box2dlights) gives us an easy way to implement lighting. We are using it for creating a cozy ambient atmosphere in the game. Below is a screenshot of some rudimentary torchlighting. 
<img width="752" alt="Screenshot 2023-11-04 at 14 22 31" src="https://github.com/SE-TINF22B6/Underwatch/assets/11832982/1f39aa99-dac2-4908-90bd-3002fae53159">

**Comment by LeonmAndreesen** [comment]
I am thrilled on your game. Looks really really cool. From what the screenshot shows you are  on a good way on implementing. In comparison to other teams you are making really fast progress. Additionally i like your models. You used a template that really enhances your models. They are really good to understand at first sight.
